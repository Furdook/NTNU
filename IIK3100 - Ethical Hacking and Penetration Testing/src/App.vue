<script setup lang="ts">
import HeaderNavigation from '@/components/header-navigation.vue'
import LoginForm from '@/components/login-form.vue'
</script>

<template>
  <div class="flex h-screen w-screen items-center justify-center">
    <header-navigation />
    <main class="w-full max-w-[540px] pb-16">
      <h2 class="mx-auto w-fit text-[1.75rem] font-medium text-heading">
        Login
      </h2>
      <a
        href=""
        class="mb-4 w-full text-center text-link hover:text-link-hover hover:underline"
        >Don't have an account? Sign up</a
      >
      <login-form />
      <a
        href=""
        class="mt-2 w-full text-center text-link hover:text-link-hover hover:underline"
        >Forgot password?</a
      >
    </main>
  </div>
</template>
