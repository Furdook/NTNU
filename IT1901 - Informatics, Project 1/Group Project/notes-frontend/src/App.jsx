import React from 'react';
import ReactDOM from 'react-dom/client';
import Notes from './components/Notes';

import './index.css';

function App() {

  return (
    <React.StrictMode>
      <main className='bg-neutral-900'>
        <Notes/>
      </main>
    </React.StrictMode>
  );
}

export default App;
