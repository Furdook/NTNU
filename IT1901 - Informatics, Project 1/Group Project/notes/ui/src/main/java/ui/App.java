package ui;

import java.io.IOException;

import core.Note;
import javafx.application.Application;
import javafx.fxml.FXMLLoader;
import javafx.scene.Parent;
import javafx.scene.Scene;
import javafx.stage.Stage;

/**
 * JavaFX App.
 */
public class App extends Application {

    private static Scene scene;
    private static Note selectedNote;

    /**
     * Starts the application.
     *
     * @param stage
     * @throws IOException
     */
    @Override
    public void start(Stage stage) throws IOException {

        setScene();
        stage.setScene(scene);
        stage.show();
    }

    /**
     * Set scene for App.
     *
     * @return scene
     * @throws IOException if
     */
    public static Scene setScene() throws IOException {
        scene = new Scene(loadFXML("Notes"));
        return scene;
    }

    /**
     * Changes the scene to the one specified by the file.
     *
     * @param file the name of the file to change the scene to
     * @throws IOException
     */
    public static void changeScene(String file) throws IOException {
        scene.setRoot(loadFXML(file));
    }

    /**
     * Loads the .fxml file specified in param
     *
     * @param file the name of the file to load
     * @return the loaded file
     * @throws IOException
     */
    private static Parent loadFXML(String file) throws IOException {
        FXMLLoader fxmlLoader = new FXMLLoader(App.class.getResource(file + ".fxml"));
        return fxmlLoader.load();
    }

    /**
     * Sets the selected note to the one specified in param (Clicked in the
     * view).
     *
     * @param note
     */
    protected static void setNote(Note note) {
        selectedNote = note;
    }

    /**
     * Returns the selected note.
     *
     * @return the selected note
     */
    protected static Note getNote() {
        return selectedNote;
    }

    /**
     * Launches the application.
     *
     * @param args
     */
    public static void main(String[] args) {
        launch();
    }
}
