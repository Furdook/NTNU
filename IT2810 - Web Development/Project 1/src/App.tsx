import { useQuery } from "@tanstack/react-query";
import Artist from "./components/Artist";
import "./styles.css";
import { Artist as A } from "./types";
import { useEffect, useState } from "react";
import Like from "./components/Like";
import { useNavigate } from "react-router";
import Filters from "./components/Filters";

function App() {
  const navigate = useNavigate();
  const { isLoading, data } = useQuery({
    queryKey: ["getAllArtists"],
    queryFn: () =>
      fetch(
        "https://musicbrainz.org/ws/2/artist/?query=area:norway&fmt=json"
      ).then((response) => response.json()),
  });

  const [currentArtist, setCurrentArtist] = useState<A>();
  const [artists, setArtists] = useState(data);

  useEffect(() => {
    if (!isLoading) {
      setArtists(data?.artists);
      setCurrentArtist(data?.artists[0]);
    }
  }, [data, isLoading]);

  return (
    <main>
      <Filters
        setCurrentArtist={setCurrentArtist}
        setArtists={setArtists}
        artists={artists}
        data={data}
      />
      <button
        id="lastArtist"
        className="buttons"
        onClick={() => {
          let currentIndex = artists.indexOf(currentArtist);
          if (currentIndex === 0) currentIndex = artists.length;
          setCurrentArtist(artists[currentIndex - 1]);
        }}
      >
        ⬅️
      </button>
      {!isLoading && <Artist {...currentArtist!} />}
      <button
        id="nextArtist"
        className="buttons"
        onClick={() => {
          let currentIndex = artists.indexOf(currentArtist);
          if (currentIndex === artists.length - 1) currentIndex = -1;
          setCurrentArtist(artists[currentIndex + 1]);
        }}
      >
        ➡️
      </button>
      <Like {...currentArtist!} />
      <button
        className="fav"
        id="view"
        onClick={() => {
          navigate("./likes");
        }}
      >
        View Likes
      </button>
    </main>
  );
}

export default App;
