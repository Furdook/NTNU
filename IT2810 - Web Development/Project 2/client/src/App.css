#routesWrapper {
  padding-top: 80px;
}
