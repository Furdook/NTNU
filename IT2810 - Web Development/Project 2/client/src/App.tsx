import "./App.css";
import { BrowserRouter as Router, Route, Routes } from "react-router-dom";
import BarNav from "./components/BarNav";
import NewPosting from "./views/NewPosting";
import Profile from "./views/Profile";
import Results from "./views/Results";
import Landing from "./views/Landing";
import PostingPage from "./views/PostingPage";
import NewProfile from "./views/NewProfile";
import Login from "./views/Login";

function App() {
  return (
    <Router basename="project2/">
      <BarNav />
      <div id="routesWrapper">
        <Routes>
          <Route path="/results/:filters" element={<Results />} />
          <Route path="/new-posting" element={<NewPosting />} />
          <Route path="/profile" element={<Profile />} />
          <Route path="*" element={<Landing />} />
          <Route path="/posting/:id" element={<PostingPage />} />
          <Route path="/new-profile" element={<NewProfile />} />
          <Route path="/login" element={<Login />} />
        </Routes>
      </div>
    </Router>
  );
}

export default App;
