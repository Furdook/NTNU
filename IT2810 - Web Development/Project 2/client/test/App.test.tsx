import React from "react";
import renderer from "react-test-renderer";
import App from "../src/App";
import { expect, it } from "vitest";

it("App renders correctly", () => {
  const tree = renderer.create(<App />).toJSON();
  expect(tree).toMatchSnapshot();
});
