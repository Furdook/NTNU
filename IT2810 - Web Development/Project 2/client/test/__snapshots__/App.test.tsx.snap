// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App renders correctly 1`] = `null`;

exports[`renders correctly 1`] = `null`;
